cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/transaction/stream/StreamAdminTest.java

```diff
@@ -74,19 +111,142 @@ public void testCreateExist() throws Exception {
     streamAdmin.create(otherStreamId);
     Assert.assertTrue(streamAdmin.exists(otherStreamId));
+
+    // the user should be able to drop the stream, they had created
+    streamAdmin.drop(otherStreamId);
+    Assert.assertFalse(streamAdmin.exists(otherStreamId));
+
+    // revoke the permission for the user on the stream in foo_namespace
+    revokeAndAssertSuccess(streamId.toEntityId(), USER, ImmutableSet.of(Action.ADMIN, Action.ALL));
+
+    try {
+      streamAdmin.drop(streamId);
+      Assert.fail("User should not be able to delete a stream in this namespace.");
+    } catch (UnauthorizedException e) {
+      // expected
+    }
+
+    // grant WRITE permission to the user but they still should not be able to drop the stream
+    grantAndAssertSuccess(streamId.toEntityId(), USER, ImmutableSet.of(Action.WRITE));
+
+    try {
+      streamAdmin.drop(streamId);
+      Assert.fail("User should not be able to delete a stream with only Write Action access.");
+    } catch (UnauthorizedException e) {
+      // expected
+    }
+
+    // grant admin access and the user should then be able to drop the stream
+    grantAndAssertSuccess(streamId.toEntityId(), USER, ImmutableSet.of(Action.ADMIN));
+    streamAdmin.drop(streamId);
+    Assert.assertFalse(streamAdmin.exists(streamId));
+  }
+
+  @Test
+  public void testConfigAndTruncate() throws Exception {
+    StreamAdmin streamAdmin = getStreamAdmin();
+    grantAndAssertSuccess(new NamespaceId(FOO_NAMESPACE), USER, ImmutableSet.of(Action.WRITE));
+    Id.Stream stream = Id.Stream.from(FOO_NAMESPACE, "stream");
+
+    streamAdmin.create(stream);
+    Assert.assertTrue(streamAdmin.exists(stream));
+    writeEvent(stream);
+
+    // Getting config / properties should work
+    streamAdmin.getConfig(stream);
+    streamAdmin.getProperties(stream);
+
+    // Now revoke access to the user to the stream
+    revokeAndAssertSuccess(stream.toEntityId(), USER, ImmutableSet.of(Action.ALL));
+    streamAdmin.getConfig(stream);
+
+    try {
+      streamAdmin.getProperties(stream);
+      Assert.fail("USer should not be able to get the properties.");
+    } catch (UnauthorizedException e) {
+      // expected
+    } catch (IOException e) {
+      Assert.assertTrue(e.getCause() instanceof UnauthorizedException);
+    }
+
+    // read action should be enough to get the stream config
+    grantAndAssertSuccess(stream.toEntityId(), USER, ImmutableSet.of(Action.READ));
+    streamAdmin.getConfig(stream);
+    StreamProperties properties = streamAdmin.getProperties(stream);
+
+    try {
+      streamAdmin.updateConfig(stream, properties);
+      Assert.fail("User should not be able to update the config with just READ permissions.");
+    } catch (UnauthorizedException e) {
+      // expected
+    }
+
+    // This call bypasses the stream handler and thus authorization is not checked for this call and so write
+    // to stream will succeed
+    writeEvent(stream);
```

Comment:
yes, write is to test truncate. Yes, write access alone is not sufficient to do truncate/drop.